src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```diff
@@ -0,0 +1,357 @@
+                    }
+            }
+
+
+            deltaSetItem.TransientInstanceAnnotationContainer = changedObj.TransientInstanceAnnotationContainer;
+            deltaSetItem.TransientInstanceAnnotationContainer.AddResourceAnnotation("Core.DataModificationException", dataModificationExceptionType);
+
+            Contract.Assert(deltaSetItem != null);
+
+            return deltaSetItem;
+        }
+
+        private IDeltaSetItem CreateEntityObjectforFailedOperation(Delta<TStructuralType> changedObj, TStructuralType originalObj)
+        {
+            Type type = typeof(Delta<>).MakeGenericType(_clrType);
+
+            Delta<TStructuralType> deltaObject = Activator.CreateInstance(type, _clrType, _clrType.GetProperties().Select(x=>x.Name), null,
+                changedObj.InstanceAnnotationsPropertyInfo) as Delta<TStructuralType>;
+
+            SetProperties(originalObj, deltaObject);
+
+            if (deltaObject.InstanceAnnotationsPropertyInfo != null) {
+
+                object instAnnValue;
+                changedObj.TryGetPropertyValue(deltaObject.InstanceAnnotationsPropertyInfo.Name, out instAnnValue);
+                IODataInstanceAnnotationContainer instanceAnnotations = instAnnValue as IODataInstanceAnnotationContainer;
+
+                if (instanceAnnotations != null)
+                {
+                    deltaObject.TrySetPropertyValue(deltaObject.InstanceAnnotationsPropertyInfo.Name, instanceAnnotations);
+                }
+            }
+
+            return deltaObject;
+        }
+
+        private void SetProperties(TStructuralType originalObj, Delta<TStructuralType> edmDeltaEntityObject)
+        {
+            foreach (string property in edmDeltaEntityObject.GetChangedPropertyNames())
+            {
+                edmDeltaEntityObject.TrySetPropertyValue(property, _clrType.GetProperty(property).GetValue(originalObj));
+            }
+
+            foreach (string property in edmDeltaEntityObject.GetUnchangedPropertyNames())
+            {
+                edmDeltaEntityObject.TrySetPropertyValue(property, _clrType.GetProperty(property).GetValue(originalObj));
+            }
+        }
+
+        private DeltaDeletedEntityObject<TStructuralType> CreateDeletedEntityForFailedOperation(Delta<TStructuralType> changedObj)
+        {
+            Type type = typeof(DeltaDeletedEntityObject<>).MakeGenericType(changedObj.ExpectedClrType);
+
+            DeltaDeletedEntityObject<TStructuralType> deletedObject = Activator.CreateInstance(type, true, changedObj.InstanceAnnotationsPropertyInfo) as DeltaDeletedEntityObject<TStructuralType>;
+
+            foreach (string property in changedObj.GetChangedPropertyNames())
+            {
+                SetPropertyValues(changedObj, deletedObject, property);
+            }
+
+            foreach (string property in changedObj.GetUnchangedPropertyNames())
+            {
+                SetPropertyValues(changedObj, deletedObject, property);
+            }
```

Comment:
why does this not use the same logic as for an changed object (i.e., the SetProperties method?)

Reply:
In one place we are copying between 2 delta<T>s vs in another case its between a T and delta<T>